### ISSUE TITLE
Inverse extract_shape?

### ISSUE THREAD

**Issue by TomasTorsvik** [ESMValGroup/ESMValCore#1027]
Hi ESMValCore developers,
thanks for providing this very useful toolset!

**Is your feature request related to a problem? Please describe.**
I am wondering how to best extract an inverse selection of the extract_shape output. Specifically, I'm using extract_shape to select all data points inside Exclusive Economic Zones (EEZ) in the ocean, and would also like to do analysis on all ocean areas remaining outside the EEZ. If I provide non-EEZ as a separate shapefile to extract_shape, I guess the grid points along the edges would be included in both analysis sets(?).
I guess the extract_shape should only return a single cube, so one solution I could think of would be to implement a method "exclude" that does the opposite of "contains", but maybe there are more elegant solutions?

**Would you be able to help out?**
Would you have the time and skills to implement the solution yourself?

I'm not very familiar with the iris cube structure. I imagine a solution could be along the line of

if method=='exclude':
   cube = cube - cubelist.merge_cube()
else:
   cube = cubelist.merge_cube()

**Comment by valeriupredoi** [comment]
good idea @TomasTorsvik :beer: Unfortunately it wouldn't be as simple as a cube subtraction since that does an actual data subtraction between two cubes with the same dimensions but a mask should be used instead. Let me assign myself to this and I will cook something up :+1:

**Comment by valeriupredoi** [comment]
I just realized we already have a PR in the pipes that does shape extraction based on user-declared ID's - see https://github.com/ESMValGroup/ESMValCore/pull/764 - I think this takes care of your used case since you can specify which shapes you want to extract, and if you don't want one in particular, then don't list it in your requested shapes? Am I correct in my logic?

**Comment by TomasTorsvik** [comment]
I think #764 addresses a different issue, if you only want to use some polygons from the shapefile for data extraction. What I would like to do is to use the shapefile polygons as a mask instead of as data selection regions. Now I'm thinking that maybe it's cleaner to do this as a "mask" step instead of an "extract_shape" step. It seems there is already a function for masking based on Natural Earth shapefiles
https://github.com/ESMValGroup/ESMValCore/blob/9f603be2d726e99fcf069ad747ddc8a73a24e89a/esmvalcore/preprocessor/_mask.py#L309
Can this be used for user defined masking as well?

**Comment by valeriupredoi** [comment]
@JaroCamphuijsen maybe you have a bit of time in the very near future look into this, mate? I am tied up with a whole lot of other stuffs at the moment and could look at it next week, but if you feel like writing some code in the meantime that'd be very cool :beer:

**Comment by JaroCamphuijsen** [comment]
Maybe it would be easiest to just add some more methods of shapely.vectorized in our _get_shape function. https://github.com/ESMValGroup/ESMValCore/blob/ebb3a8dbf76bff60443449c6a5674381bc4b6639/esmvalcore/preprocessor/_area.py#L444

I think if we would add "intersects" (https://shapely.readthedocs.io/en/latest/manual.html#object.intersects) you could accomplish what was asked for with the inverted shapefile. 

Or we could add "disjoint", this would be the exact opposite of "contains". (https://shapely.readthedocs.io/en/latest/manual.html#object.disjoint)
